tests/utils/managed_utils/client.rs

```diff
@@ -175,12 +173,16 @@ impl Client {
         let action_results = match action_type {
             ActionType::Commit => {
                 let (messages, welcome) = group.add_members(&self.key_store, key_packages)?;
-                (messages, Some(welcome))
+                (vec![messages], Some(welcome))
+            }
+            ActionType::Proposal => {
+                let mut messages = Vec::new();
+                for key_package in key_packages {
```

Comment:
This looks like it should be a `for_each()` with `collect()`.

Reply:
Good idea! I'm not sure how I should handle the potential error thrown by `propose_add_member` in the `for_each` closure, though. Same in the other code below.

Reply:
Ah, I didn't see the error there. Then let's leave it like this for now.